internal/scheduler/scheduling_algo.go

```diff
@@ -483,9 +489,21 @@ func (l *FairSchedulingAlgo) schedulePool(
 			l.limiterByQueue[queue] = queueLimiter
 		}
 
-		queueLimiter.SetLimitAt(now, rate.Limit(l.schedulingConfig.MaximumPerQueueSchedulingRate))
+		queueContextDemand := demand.AggregateByResource()
 
-		if err := sctx.AddQueueSchedulingContext(queue, weight, allocatedByPriorityClass, demand.AggregateByResource(), cappedDemand.AggregateByResource(), queueLimiter); err != nil {
+		if !fsctx.schedulingStatusByQueue[queue] {
+			queueLimiter.SetLimitAt(now, rate.Limit(float64(0)))
+			queueLimiter.SetBurstAt(now, 0)
+
+			// Queued jobs should not be considered for paused queues, so demand := running
```

Comment:
can't we just take this into account when we calculate demand? Rather than calculating demand and then changing that later on